src/modules/auth/auth.service.ts

```diff
@@ -1,50 +1,84 @@
-import { AccountType } from './enums/account-type.enum';
-import { Payload } from './interfaces/payload.interface';
-import { UserToken } from './interfaces/user-token.interface';
-import { JwtStrategy } from './strategies/jwt.strategy';
-
+import { JwtPayload } from './interfaces/jwt-payload.interface';
+import { JwtSignature } from './interfaces/jwt-signature.interface';
+import { AuthStrategy } from './strategies/auth.strategy';
+import { UserService } from '../user/user.service';
+import { IUser } from '../user/interfaces/user.interface';
+import { IRegisterUser } from './interfaces/register.interface';
+import { AuthUtil } from 'src/shared/utils/auth.util';
+import { MailerService } from '@nestjs-modules/mailer';
+import { MailStrategy } from './strategies/mail.strategy';
+import { AuthBusinessExceptions } from './exceptions/auth-business.exceptions';
 @Injectable()
 export class AuthService {
   constructor(
-    private readonly userRepository: UserRepository,
-    private readonly jwtStrategy: JwtStrategy,
+    private authStrategy: AuthStrategy,
+    private mailStrategy: MailStrategy,
+    private mailerService: MailerService,
+    private userService: UserService,
   ) {}
 
-  async login(userLoginDto: UserLoginDto): Promise<UserToken> {
-    const { email, password } = userLoginDto;
-    const payload = await this.validate(email, password);
-    delete userLoginDto.password;
-    const userToken = await this.jwtStrategy.getUserToken(payload);
-    return userToken;
+  async register(userData: IRegisterUser): Promise<IUser> {
+    const password = AuthUtil.hash(userData.password);
+
+    const signature = await this.mailStrategy.sign({
+      email: userData.email,
+    });
+
+    const createdUser = await this.userService.createUser({
+      firstName: userData.firstName,
+      lastName: userData.lastName,
+      email: userData.email,
+      password,
+      accessToken: signature.accessToken,
+      birthDate: userData.birthDate,
+      termsAccepted: userData.termsAccepted,
+    });
+
+    // await this.mailerService.sendMail({
+    //   to: createdUser.email,
+    //   subject: 'Confirmação de Cadastro',
+    //   template: 'email-confirmation',
+    //   context: {
+    //     username: createdUser.firstName,
+    //     url: `${process.env.EMAIL_CONFIRMATION_URL}?token=${access_token}`,
+    //   },
+    // });
+
+    return createdUser;
   }
 
-  async validate(email: string, password: string): Promise<Payload> {
-    const user = await this.userRepository.findByEmail(email);
-    if (user === null) throw new UnauthorizedException('Credenciais inválidas');
-    if (user.isEmailConfirmed === false) {
-      throw new UnauthorizedException('Usuário não verificado');
+  async login(userLoginDto: UserLoginDto): Promise<JwtSignature> {
+    const user = await this.userService.getUserByEmail(userLoginDto.email);
+
+    if (!user || !AuthUtil.verify(userLoginDto.password, user.password))
```

Comment:
Essa classe poderia ter um nome melhor, como por exemplo, somente "Password" ou algo parecido, ja que sua responsabilidade é lidar somente com senhas. Em outros casos, poderia ser algo relacionado a criptografia se for além do tratamento de senhas

Reply:
Irei considerar que possa ir além de apenas senhas.